File: SchoolWebApp-API/Project.Core/Entities/CBE/Values/StudentValueScore.cs
using SchoolWebApp.Core.Entities.Class;
using SchoolWebApp.Core.Entities.Shared;
using SchoolWebApp.Core.Entities.Students;

namespace SchoolWebApp.Core.Entities.CBE.Values
{
    public class StudentValueScore: Base
    {
        public int StudentId { get; set; }
        public Student? Student { get; set; } = null;
        public int ValueId { get; set; }
        public Value? Value { get; set; } = null;
        public int SessionId { get; set; }
        public Session? Session { get; set; } = null;
        public int ValueScoreId { get; set; }
        public ValueScore? ValueScore { get; set; }
        public string? Description { get; set; } = null;
    }
}

File: SchoolWebApp-Client/src/app/cbe/values/models/student-value-score.ts
import {ResourceModel} from '@/core/models/ResourceModel';

export class StudentValueScore extends ResourceModel<StudentValueScore> {
    public studentId?: number;
    public student?: any;
    public valueId?: number;
    public value?: any;
    public sessionId?: number;
    public session?: any;
    public valueScoreId?: number;
    public valueScore?: any;
    public description?: string;

    constructor(model?: Partial<StudentValueScore>) {
        super(model);
    }
}
